src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Egress/S3/MultiPartUploadStreamTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.Egress.S3;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests.Egress.S3
+{
+    public class MultiPartUploadStreamTests
+    {
+
+        private readonly InMemoryS3 _s3 = new InMemoryS3();
+
+        private IEnumerable<byte[]> WithBytesReturned(int totalBytes)
+        {
+            while(totalBytes > 0)            
+            {
+                var length = Random.Shared.Next(1, Math.Min(totalBytes, 1024));
+                var bytes = new byte[length];
+                Random.Shared.NextBytes(bytes);
+                yield return bytes;
+                totalBytes -= length;
+            }
+        }
+
+        [Fact]
+        public async Task ItShouldWriteMultipleParts()
+        {
+            var uploadId = (await _s3.InitiateMultipartUploadAsync("bucket", "key")).UploadId;
+            await using var stream = new MultiPartUploadStream(_s3, "bucket", "key", uploadId, 1024);
```

Comment:
Can you add another test that make sure that a buffer size greater than the minimum size works as expected? Something like 6 MB and write 8 MB to the stream (thus two parts, one 6MB, the other 2 MB).